src/components/dialogs/inputs/input-hooks.js

```diff
@@ -864,3 +736,81 @@ export const useCSVReader = ({ label, header }) => {
     }, [selectedFile, intl, header]);
     return [selectedFile, setSelectedFile, field, fileError];
 };
+
+export const useRadioValue = ({
+    label,
+    possibleValues = [],
+    defaultValue,
+    id,
+    validation = {},
+    inputForm,
+    doTranslation = true,
+}) => {
+    const [value, setValue] = useState(defaultValue);
+    const intl = useIntl();
+
+    useEffect(() => {
+        // Updates the component when the correct default value is given by the parent component.
+        if (defaultValue !== undefined && defaultValue.length > 0) {
+            setValue(defaultValue);
+        }
+    }, [defaultValue]);
+
+    useEffect(() => {
+        function validate() {
+            return true;
+        }
+
+        inputForm.addValidation(id ? id : label, validate);
+    }, [label, validation, inputForm, value, id]);
+
+    const handleChangeValue = useCallback((event) => {
+        setValue(event.target.value);
+    }, []);
+
+    const field = useMemo(() => {
+        return (
+            <FormControl>
+                {label && (
+                    <FormLabel id={id ? id : label}>
+                        {intl.formatMessage({ id: label })}
+                    </FormLabel>
+                )}
+                <RadioGroup
+                    row
+                    aria-labelledby={id ? id : label}
+                    value={value ?? defaultValue}
+                    onChange={handleChangeValue}
+                >
+                    {possibleValues.map((value) => (
+                        <FormControlLabel
+                            value={value}
+                            control={<Radio />}
+                            label={
+                                doTranslation
+                                    ? intl.formatMessage({ id: value })
+                                    : value
+                            }
+                        />
+                    ))}
+                </RadioGroup>
+            </FormControl>
+        );
+    }, [
+        intl,
+        label,
+        handleChangeValue,
+        id,
+        defaultValue,
+        doTranslation,
+        possibleValues,
+        value,
+    ]);
+
+    useEffect(
```

Comment:
You already have a useEffect line 752 which is proc by `defaultValue` which do the exact same action. Why not using only one UseEffect ?

Reply:
Nice catch, the second one is an error. I removed it.